### PR TITLE
[BUGFIX] Pandas ID/PK - bugfix for column name and update tests

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
Changes proposed in this pull request:
- Bugfix for evaluating JSON tests for ID/PK 
- Update JSON tests that define `unexpected_index_list`, so they are up to speed with ID/PK output

### Definition of Done
- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/self_check/util.py [hunk 1]
```diff
@@ -2873,7 +2873,7 @@ def evaluate_json_test_v3_api(  # noqa: C901 - 16
                 runtime_kwargs = {
                     "result_format": {
                         "result_format": "COMPLETE",
-                        "unexpected_index_columns": ["pk_index"],
+                        "unexpected_index_column_names": ["pk_index"],
```
Comment (Shinnnyshinshin) [inline great_expectations/self_check/util.py:2876]: 😓

File: great_expectations/self_check/util.py [hunk 2]
```diff
@@ -3072,9 +3072,10 @@ def check_json_test_result(  # noqa: C901 - 52
                 elif isinstance(data_asset, (SqlAlchemyBatchData, SparkDFBatchData)):
                     pass
                 else:
-                    assert (
-                        result["result"]["unexpected_index_list"] == value
-                    ), f"{result['result']['unexpected_index_list']} != {value}"
+                    if pk_column:
+                        assert (
+                            result["result"]["unexpected_index_list"] == value
+                        ), f"{result['result']['unexpected_index_list']} != {value}"
```
Comment (Shinnnyshinshin) [inline great_expectations/self_check/util.py:3076]: This ensures that V2 Expectations do not do the `unexpected_index_list` assertion

File: great_expectations/self_check/util.py [hunk 3]
```diff
@@ -3000,6 +3003,17 @@ def check_json_test_result(  # noqa: C901 - 52
         result = result.to_json_dict()
         for key, value in test["output"].items():
             # Apply our great expectations-specific test logic
+            if v2_or_v3_api is "v2" and key in [
```
Comment (Shinnnyshinshin) [inline great_expectations/self_check/util.py]: these metrics are not implemented for V2 `MultiColumnMapExpectations`

File: great_expectations/self_check/util.py [hunk 4]
```diff
@@ -3185,7 +3200,8 @@ def check_json_test_result(  # noqa: C901 - 52
                         ],
                         value["tail_weights"],
                     )
-
+            elif key == "exact_match_out":
```
Comment (anthonyburdi) [inline great_expectations/self_check/util.py]: I'm having trouble following why this change is needed - could you explain? Is it because `exact_match_out` configures the test and should not be checked, and this change makes that explicit?

File: tests/test_definitions/multicolumn_map_expectations/expect_select_column_values_to_be_unique_within_record.json [hunk 5]
```diff
@@ -126,26 +126,19 @@
       },
       "out": {
         "unexpected_list": [{"a": 1.0, "b": 1.0}, {"a": 1.0, "b": 1.0}, {"a": 1.0, "b": 1.0}, {"a": 2.0, "b": 2.0}, {"a": 2.0, "b": 2.0}],
-        "unexpected_index_list": [0,2,4,5,7],
+        "unexpected_index_list": [{"a": 1.0, "pk_index": 0, "b": 1.0}, {"a": 1.0, "pk_index": 2, "b": 1.0}, {"a": 1.0, "pk_index": 4, "b": 1.0}, {"a": 2.0, "pk_index": 5, "b": 2.0}, {"a": 2.0, "pk_index": 7, "b": 2.0}],
         "success": false
       }
     },{
       "title": "unexpected_values_exact_match_out_with_index_list",
-      "exact_match_out" : true,
```
Comment (anthonyburdi) [inline tests/test_definitions/multicolumn_map_expectations/expect_select_column_values_to_be_unique_within_record.json:134]: Why was this changed to false?

File: tests/test_definitions/multicolumn_map_expectations/expect_compound_columns_to_be_unique.json [hunk 6]
```diff
@@ -240,21 +286,14 @@
       }
     },{
       "title": "unexpected_values_exact_match_out_with_unexpected_index_list",
-      "exact_match_out" : true,
```
Comment (anthonyburdi) [inline tests/test_definitions/multicolumn_map_expectations/expect_compound_columns_to_be_unique.json:243]: Why was this changed?

**Review by anthonyburdi — COMMENTED** [review]
Just a few questions to help understand. Also posted a question in slack to understand motivation for this change.

File: tests/test_definitions/multicolumn_map_expectations/expect_select_column_values_to_be_unique_within_record.json (hunk 5 above)
Comment (Shinnnyshinshin) [reply tests/test_definitions/multicolumn_map_expectations/expect_select_column_values_to_be_unique_within_record.json:134]: This is because the JSON tests handle both v2 and v3 expectations. 
* `exact_match_out` ensures the entire dictionary matches expectations, but this can't be the case. 
* V3 will contain ID/PK related parameters in the `unexpected_index_list` while V2 will not

File: tests/test_definitions/multicolumn_map_expectations/expect_compound_columns_to_be_unique.json (hunk 6 above)
Comment (Shinnnyshinshin) [reply tests/test_definitions/multicolumn_map_expectations/expect_compound_columns_to_be_unique.json:243]: This is because the JSON tests handle both v2 and v3 expectations. 
* `exact_match_out` ensures the entire dictionary matches expectations, but this can't be the case. 
* V3 will contain ID/PK related parameters in the `unexpected_index_list` while V2 will not

File: great_expectations/self_check/util.py (hunk 3 above)
Comment (Shinnnyshinshin) [reply great_expectations/self_check/util.py]: no longer having to do this split after distinguishing between the `pandas_v2_api` and `pandas_v3_api` tests :)

File: great_expectations/self_check/util.py (hunk 4 above)
Comment (Shinnnyshinshin) [reply great_expectations/self_check/util.py]: no longer having to do this check after splitting the pandas_v2_api and pandas_v3_api tests :)

File: tests/test_definitions/multicolumn_map_expectations/expect_select_column_values_to_be_unique_within_record.json [hunk 7]
```diff
@@ -126,13 +126,14 @@
...
+        "unexpected_index_list": [{"a": 1.0, "pk_index": 0, "b": 1.0}, {"a": 1.0, "pk_index": 2, "b": 1.0}, {"a": 1.0, "pk_index": 4, "b": 1.0}, {"a": 2.0, "pk_index": 5, "b": 2.0}, {"a": 2.0, "pk_index": 7, "b": 2.0}],
         "success": false
       }
-    },{
+    },
+      {
       "title": "unexpected_values_exact_match_out_with_index_list",
       "exact_match_out" : true,
-      "only_for": ["pandas"],
+      "only_for": ["pandas_v2_api"],
```
Comment (Shinnnyshinshin) [inline tests/test_definitions/multicolumn_map_expectations/expect_select_column_values_to_be_unique_within_record.json:136]: This now does the split between the `pandas_v2_api` and `pandas_v3_api`

File: tests/test_definitions/multicolumn_map_expectations/expect_compound_columns_to_be_unique.json [hunk 8]
```diff
@@ -239,9 +285,9 @@
         "success": false
       }
     },{
-      "title": "unexpected_values_exact_match_out_with_unexpected_index_list",
+      "title": "unexpected_values_exact_match_out_with_unexpected_index_list_pandas_v2_api",
       "exact_match_out" : true,
-      "only_for": ["pandas"],
+      "only_for": ["pandas_v2_api"],
```
Comment (Shinnnyshinshin) [inline tests/test_definitions/multicolumn_map_expectations/expect_compound_columns_to_be_unique.json:290]: Similar story here `pandas_v2_api` and `pandas_v3_api`

File: tests/test_definitions/multicolumn_map_expectations/expect_compound_columns_to_be_unique.json [hunk 9]
```diff
@@ -338,6 +384,142 @@
           "unexpected_index_list": [0,1,2,3,4,5,6,7,8]
         }
       }
+    },{
+      "title": "unexpected_values_exact_match_out_with_unexpected_index_list_pandas_v3",
+      "exact_match_out" : true,
+      "only_for": ["pandas_v3_api"],
```
Comment (Shinnnyshinshin) [inline tests/test_definitions/multicolumn_map_expectations/expect_compound_columns_to_be_unique.json:390]: `exact_match_out` is maintained as `true`

File: tests/test_definitions/multicolumn_map_expectations/expect_select_column_values_to_be_unique_within_record.json [hunk 10]
```diff
@@ -209,6 +210,90 @@
           ]
         }
       }
+    },{
+      "title": "unexpected_values_exact_match_out_with_index_list_pandas_v3_api",
+      "exact_match_out" : true,
```
Comment (Shinnnyshinshin) [inline tests/test_definitions/multicolumn_map_expectations/expect_select_column_values_to_be_unique_within_record.json:215]: `exact_match_out` is maintained as `true`

**Review by anthonyburdi — APPROVED** [review]
LGTM! Thank you for making those changes.
